include/BSMPT/bounce_solution/action_calculation.h

```diff
@@ -0,0 +1,821 @@
+// Copyright (C) 2024 Lisa Biermann, Margarete Mühlleitner, Rui Santos, João
+// Viana
+//
+// SPDX-FileCopyrightText: 2024 Lisa Biermann, Margarete Mühlleitner, Rui
+// Santos, João Viana
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+/**
+ * @file
+ */
+
+// Remove warning from <experimental/filesystem>
+#define _LIBCPP_NO_EXPERIMENTAL_DEPRECATION_WARNING_FILESYSTEM
+
+#include <BSMPT/utility/const_velocity_spline.h>
+#include <BSMPT/utility/utility.h>
+#include <Eigen/Dense>
+#include <numeric>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+using Eigen::MatrixXd;
+using Eigen::VectorXd;
+
+#ifndef INCLUDE_BSMPT_GravitationalWaves_GWUtils_ActionCalculation_H_
+#define INCLUDE_BSMPT_GravitationalWaves_GWUtils_ActionCalculation_H_
+
+/**
+ * @brief This classes calculates the Bounce action of the potential with a set
+ * temperature
+ */
+
+namespace BSMPT
+{
+
+class BounceActionInt
+{
+private:
+  // These two variables check if the method undershot and overshot at least
+  // once. If only one type of convergence was achieved it is an indication that
+  // no solution has been found.
+  /**
+   * @brief Records if overshoot/undershoot method undershots at least once
+   */
+  bool undershot_once = false;
+  /**
+   * @brief Records if overshoot/undershoot method overshots at least once
+   */
+  bool overshot_once = false;
+
+  /**
+   * @brief Value of d2Vdl^2 near the true vacuum
+   *
+   */
+  double TrueVacuumHessian;
+
+  /**
+   * @brief Store the value of backwards propagation
+   *
+   */
+  double Initial_lmin;
+
+  /**
+   * @brief Potential at the false vacuum in the unshifted potential (in the
+   * shifted Vfalse = 0)
+   *
+   */
+  double Vfalse;
+
+  /**
+   * @brief l0 - Initial_lmin for solutions starting very near the true vacuum
+   *
+   */
+  double l0_Initial_lmin;
+
+  /**
+   * @brief True if path deformation reached the desired results without solving
+   * the 1D equation one more time.
+   *
+   */
+  bool PathDeformationConvergedWithout1D = false;
+
+public:
+  /**
+   * @brief Dimension of the VEV space
+   *
+   */
+  int dim = -1;
+
+  /**
+   * @brief Action calculated by the class
+   *
+   * * = -1 if not yet calculated.
+   * * = -2 if 1D integration did not work.
+   * * = -3 if the path deformation did not converged in time.
+   * * = -4 if the path deformation crashed.
+   * * = -5 if the false vacuum in not a minimum.
+   * * = -6 path deformation created a path with V(True) > V(False) (even with
+   * backwards propagation).
+   * * = -7 1D integration never overshot/undershot
+   * * = -8 overshooting/undershooting found a negative gradient point
+   * * = -9 if the solution does not have enough points to populate the \f$
+   * l(\rho) \f$ spline
```

Comment:
This looks like it should be an enum. Are there any reasons not to make an enum out of it?

Reply:
Fixed